### Importig the libraries and the dataset

In [8]:
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

In [9]:
data=pd.read_csv('50_Startups.csv')
X=data.iloc[:,:-1].values
y=data.iloc[:,-1].values

### Encoding the Categorical Feature of "State"

In [10]:
from sklearn.compose import ColumnTransformer
from sklearn.preprocessing import OneHotEncoder

In [11]:
ct=ColumnTransformer(transformers=[(('encoder'), OneHotEncoder(),[3])], remainder='passthrough')
X=np.array(ct.fit_transform(X))

### Splitting the dataset into training and test sets

In [14]:
from sklearn.model_selection import train_test_split
X_train,X_test,y_train,y_test=train_test_split(X,y, test_size=0.2, random_state=0)

### Training the model on training set

In [15]:
from sklearn.linear_model import LinearRegression
regressor= LinearRegression()
regressor.fit(X_train,y_train)

LinearRegression()

### Predicting the Test Set Results

In [21]:
y_pred=regressor.predict(X_test)
np.printoptions(precision=2)
print(np.concatenate((y_pred.reshape(len(y_pred),1),y_test.reshape(len(y_test),1)),1))

[[103015.20159796 103282.38      ]
 [132582.27760815 144259.4       ]
 [132447.73845175 146121.95      ]
 [ 71976.09851258  77798.83      ]
 [178537.48221056 191050.39      ]
 [116161.24230167 105008.31      ]
 [ 67851.69209676  81229.06      ]
 [ 98791.73374687  97483.56      ]
 [113969.43533013 110352.25      ]
 [167921.06569551 166187.94      ]]


###  Making Single Predictions using the model

In [25]:
regressor.predict([[1,0,0,1600000,1500000,1400000]])

array([1380682.60850415])

In [27]:
print(regressor.coef_)
print(regressor.intercept_)

[ 8.66383692e+01 -8.72645791e+02  7.86007422e+02  7.73467193e-01
  3.28845975e-02  3.66100259e-02]
42467.52924853325


##### Profit=86.6×Dummy State 1−873×Dummy State 2+786×Dummy State 3+0.773×R&D Spend+0.0329×Administration+0.0366×Marketing Spend+42467.53

#### Here the dummy states are the columns representing the encoded digits representing the categorical column